File: mission-to-mars-solid/src/main/java/mars/Direction.java
package mars;

import lombok.Getter;

public enum Direction {

	NORD(new MouvementNord2D()),
	SUD(new MouvementSud2D()),
	OUEST(new MouvementOuest2D()),
	EST(new MouvementEst2D());

	@Getter
	private final Mouvement2D mouvement2D;
	
	Direction(Mouvement2D mouvement2D) {
		this.mouvement2D = mouvement2D;
	}

}

File: python/mission-to-mars-solid/src/mars/Direction.py
import enum

from src.mars.MouvementEst2D import MouvementEst2D
from src.mars.MouvementNord2D import MouvementNord2D
from src.mars.MouvementOuest2D import MouvementOuest2D
from src.mars.MouvementSud2D import MouvementSud2D


class Direction(enum.Enum):
    NORD = MouvementNord2D()
    SUD = MouvementSud2D()
    OUEST = MouvementOuest2D()
    EST = MouvementEst2D()
